src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTests.cs

```diff
@@ -17,67 +17,54 @@
-    private Dictionary<string, int> FilePathCountMap = new();
+    private const string SdkType = "sdk";
+
     private StringBuilder Differences = new();
+    private List<string> NewExclusions = new List<string>();
+    private Dictionary<string, int> FilePathCountMap = new();
+    ExclusionsHelper exclusionsHelper = new ExclusionsHelper("ArtifactExclusions.txt", nameof(ArtifactsSizeTests));
 
-    public ArtifactsSizeTests(ITestOutputHelper outputHelper) : base(outputHelper)
-    {
-        if (File.Exists(BaselineFilePath))
-        {
-            string[] baselineFileContent = File.ReadAllLines(BaselineFilePath);
-            foreach (string entry in baselineFileContent)
-            {
-                string[] splitEntry = entry.Split(':', StringSplitOptions.TrimEntries);
-                Baseline[splitEntry[0]] = long.Parse(splitEntry[1]);
-            }
-        }
-        else
-        {
-            Assert.Fail($"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
-        }
-    }
+    public ArtifactsSizeTests(ITestOutputHelper outputHelper) : base(outputHelper) {}
 
     [ConditionalFact(typeof(Config), nameof(Config.IncludeArtifactsSizeTests))]
-    public void CompareArtifactsToBaseline()
+    public void CheckZeroSizeArtifacts()
     {
-        Assert.False(string.IsNullOrWhiteSpace(Config.SourceBuiltArtifactsPath));
         Assert.False(string.IsNullOrWhiteSpace(Config.SdkTarballPath));
 
-        var tarEntries = ProcessSdkAndArtifactsTarballs();
-        ScanForDifferences(tarEntries);
-        UpdateBaselineFile();
+        ProcessTarball(Config.SdkTarballPath, SdkType);
 
-        // Must wait to report differences until after the baseline file is updated else a failure
-        // will cause the baseline file to not be updated.
+        exclusionsHelper.GenerateNewBaselineFile(updatedFileTag: null, NewExclusions);
+
+        // Wait to report differences until after the baseline file is updated. 
+        // Else a failure will cause the baseline file to not be updated.
         ReportDifferences();
     }
 
-    private Dictionary<string, long> ProcessSdkAndArtifactsTarballs()
+    private void ProcessTarball(string tarballPath, string type)
     {
         string tempTarballDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
         Directory.CreateDirectory(tempTarballDir);
 
-        Utilities.ExtractTarball(Config.SdkTarballPath!, tempTarballDir, OutputHelper);
-        Utilities.ExtractTarball(Config.SourceBuiltArtifactsPath!, tempTarballDir, OutputHelper);
+        Utilities.ExtractTarball(tarballPath, tempTarballDir, OutputHelper);
+
+        foreach (string filePath in Directory.EnumerateFiles(tempTarballDir, "*", SearchOption.AllDirectories))
+        {
+            string relativePath = filePath.Substring(tempTarballDir.Length + 1);
+            string processedPath = ProcessFilePath(relativePath);
```

Comment:
Neither of these lines need to be executed if the file size is 0. They should be moved to be inside the `if (new FileInfo(filePath).Length == 0)` condition, or the `foreach` above should enumerate a set of files which are already filtered to the ones with size 0.